R/interpolate.R

```diff
@@ -63,12 +65,22 @@ interpolate_package <- function(
   ...,
   .envir = parent.frame()
 ) {
-  path <- system.file("prompts", path, package = package)
-  interpolate_file(path, ..., .envir = .envir)
+  check_string(path)
+
+  path_pkg <- system.file("prompts", path, package = package)
```

Comment:
Oh, one bit of friction here is that `pkgload` shims `system.file()` in the global env and in the loaded package namespace, so `interpolate_package()` doesn't work with dev packages.

Where I've needed that, I've written an internal `interpolate_package()` that replicated ellmer's implementation. But it'd be nice if there were a way around this. (The Shiny team has a `system_file()` function in [Winston's staticimports package](https://github.com/wch/staticimports/blob/35ceec8d9d9429d9244aedc3ee6a1e8d62d59f79/inst/staticexports/package.R#L35-L41), but maybe there's rlang helper I haven't encountered yet?)

Reply:
Yeah, I think it's probably worth testing if package is a dev package (i.e. implement the shim ourselves).